mava/wrappers/jumanji.py

```diff
@@ -93,6 +59,84 @@ def step(
         return state, timestep
 
 
+class MultiAgentWrapper(Wrapper):
+    """
+    Multi-agent wrapper for environments with configurable reward handling.
+
+    Args:
+        env (Environment): The base environment.
+        use_individual_reward (bool): If True, the network uses the list of different rewards given.
+        aggregate_rewards (bool): If True, aggregates rewards across agents.
+        aggregate_rewards_with (str): The function for aggregating rewards ("sum" or "mean").
+    """
+
+    def __init__(
+        self,
+        env: Environment,
+        use_individual_reward: bool = False,
+        aggregate_rewards: bool = False,
+        aggregate_rewards_with: str = "sum",
+    ):
+        super().__init__(env)
+        self._use_individual_reward = use_individual_reward
+        self._aggregate_rewards = aggregate_rewards
+        self.aggregate_rewards_with = aggregate_rewards_with
+        self._num_agents = self._env.num_agents
+        # Mapping aggregation functions
+        self.aggregation_functions = {
+            "sum": jnp.sum,
+            "mean": jnp.mean,
+            # Add more functions as needed
+        }
+        self.aggregate_function = self.aggregation_functions.get(
+            aggregate_rewards_with, jnp.sum
+        )  # No error!
+
+    def aggregate_rewards(
+        self, timestep: TimeStep, observation: Observation
+    ) -> TimeStep[Observation]:
+        """Aggregate rewards across agents using the specified function."""
+        reward = jnp.repeat(self.aggregate_function(timestep.reward), self._num_agents)
+        return timestep.replace(observation=observation, reward=reward)
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep based on the specified reward handling strategy."""
+        observation = Observation(
+            agents_view=timestep.observation.agents_view,
+            action_mask=timestep.observation.action_mask,
+            step_count=jnp.repeat(timestep.observation.step_count, self._num_agents),
+        )
+
+        if self._use_individual_reward:
```

Comment:
Maybe just a comment at the top explaining that this will replace the timesteps accordingly? It is a bit hard to follow with all the returns.

Reply:
I added some details as comments and docs, should we add more or change something?